.github/workflows/publish.yaml

```diff
@@ -0,0 +1,79 @@
+name: Docker Image Publish
+
+on:
+  push:
+    tags:
+      - v*
+
+env:
+  EDGE: v1.14/alpine
+
+jobs:
+  publish:
+    name: Publish Docker Image
+    strategy:
+      fail-fast: false
+      matrix:
+        version: [v1.14]
+        os: [alpine]
+    runs-on: ubuntu-latest
+    env:
+      EDGE_TAGS: ${{ fromJSON('["","type=raw,value=edge"]')[format('{0}/{1}', matrix.version, matrix.os) == env.EDGE] }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Docker meta
+        id: docker_meta
+        uses: crazy-max/ghaction-docker-meta@v3
+        with:
+          images: fluent/fluentd-aggregator,ghcr.io/fluent/fluentd-aggregator
+          tags: |
+            type=semver,pattern={{version}}
+            type=semver,pattern={{major}}.{{minor}}
+            ${{ env.EDGE_TAGS }}
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build & push
+        id: docker_build
+        uses: docker/build-push-action@v2
+        with:
+          file: ./${{ matrix.version }}/${{ matrix.os }}/Dockerfile
+          context: ./${{ matrix.version }}/${{ matrix.os }}
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ steps.docker_meta.outputs.tags }}
+          labels: ${{ steps.docker_meta.outputs.labels }}
+
+      - name: Update Docker repo description
+        if: ${{ format('{0}/{1}', matrix.version, matrix.os) == env.EDGE }}
+        uses: peter-evans/dockerhub-description@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+          repository: fluent/fluentd-aggregator
+
+      - name: Create release
+        if: ${{ format('{0}/{1}', matrix.version, matrix.os) == env.EDGE }}
```

Comment:
These multiple if checks seem like a smell, would it be better as a separate job with just the one guard and doing both steps for release?